### DISCUSSION TITLE
DB migration error with upgrade from 6.5 to 7.1. [SOLVED]

### DISCUSSION THREAD

**Discussion by gpe5280** [ampache/ampache#4100] [General]
I'm attempting to upgrade from 6.5.0 to 7.1.1, so a lot of DB updates. I'm receiving an unspecific error with Migration700005.

```
# php bin/cli admin:updateDatabase -e
Ampache version: 7.1.1
Config version: 77
Database version: 7.0.0 Build: 004

Update Now!

Ampache\Module\System\Update\Migration\V7\Migration700005

Error
```

Is the jump from 6.5 to 7 to significant?

Update

I restored the 6.5 directory and attempted to downgrade the DB.

```
# php bin/cli admin:updateDatabase -e
Ampache version: 6.5.0
Config version: 71
Database version: 7.0.0 Build: 004
No update needed
```

No joy there either. Although, everything seems to be working fine.

**Comment by lachlan-00** [comment]
did it continue past the update and just print "Error" for that one?

Migration700005 is a last_count column and then updates the playlists so there my be an issue updating the playlists and not the database.

what it should be doing if it's a SQL error is printing the bad SQL so it seems like a different issue.

(sorry i was away on holiday for a few weeks)

If you want some help with this and can SQL query okay i can work on what the error is.

**Reply by gpe5280** [reply]
No worries about the delay. Version 6.5 seems to be running fine even with the partial DB migration.

No, the upgrade didn't move past Migration700005 and the only message was the single word: "Error." I checked through every log I could think of (syslog, apache, MySQL, Ampache) but there wasn't so much as a hint of what the error might be.

I'm fairly proficient at SQL (former Oracle DBA, ages ago) so whatever you can suggest would be great. I did briefly explore if it was possible to run the Migration700005 manually, but decided against any experimentation down that path.

**Reply by lachlan-00** [reply]
okay thank you i'll get some queries for you to test i've put one little thing in  as i think one of the queries might have been an issue.

**Reply by lachlan-00** [reply]
After your downgrade there isn't much that should affect you.

This should error if your downgrade worked. 6.5 might not have all the downgrades available though.
```
SELECT `last_count` FROM `playlist` LIMIT 1;
```

if you don't get an error, drop the column. the updates before 700005 that require a downgrade are just preferences so won't affect you.
```
ALTER TABLE `playlist` DROP COLUMN `last_count`;
```

Now i want to test this one. it's a loader that gets all the counts into the next query to insert
```
SELECT `playlist`.`id`, COUNT(`playlist_data`.`id`) AS `count` FROM `playlist` LEFT JOIN `playlist_data` ON `playlist_data`.`playlist` = `playlist`.`id` GROUP BY `playlist`.`id`;
```

I think you may have an empty playlist? or a null output here? (unless you added to it since then)

The error could be null data being input into the next query or something with the loading....
![image](https://github.com/user-attachments/assets/a8e4867b-67b8-40c6-9693-b07a0747d495)

actually that area is the most likely problem. especially if your table does have last_count column

i know it was a while ago but it didn't log anything like a runtime error?

**Reply by lachlan-00** [reply]
fbf53c761f2f72185f4059ad82c4b0f3aa498de4 does it without calling the new Playlist which i think is why you got the Error message without detail

**Reply by gpe5280** [reply]
Executed:

```
USE ampache;
SELECT `last_count` FROM `playlist` LIMIT 1;

1 row(s) returned
```

Output: 57

No error, so time to drop the column.

```
ALTER TABLE `playlist` DROP COLUMN `last_count`;

0 row(s) affected Records: 0  Duplicates: 0  Warnings: 0
```

Re-executed:

```
SELECT `last_count` FROM `playlist` LIMIT 1;

Error Code: 1054. Unknown column 'last_count' in 'field list'
```

Executed:

```
SELECT `playlist`.`id`, COUNT(`playlist_data`.`id`) AS `count` FROM `playlist` LEFT JOIN `playlist_data` ON `playlist_data`.`playlist` = `playlist`.`id` GROUP BY `playlist`.`id`;

37 row(s) returned
```

No errors or null output. I haven't change anything in the database since before the attempted upgrade - no new or modified playlists, no new audio files, etc. Neither have I seen any runtime errors or noticed any intermittent issues. I use Ampache every day for hours at a time.

Out of curiosity, I looked for the code snippet in the currently install code base (v6.5) and found it in ./src/Module/System/Update/Migration/V6/Migration600073.php

I'm also wondering if I should have attempted the DB rollback from the 7.1 code base rather than from the restored 6.5 code base.

As for the patch, I want to be clear on what I should do.

1. Download and extract the latest Ampache code (v7.1.1)
2. Patch the src/Module/System/Update/Migration/V7/Migration700005.php file
3. Re-attempt the upgrade

**Reply by lachlan-00** [reply]
Oh the last_count is in 6 as well. I missed that, dropping the column is bad then!

I'll update that migration as well to match with ea0276f720c3410d0ca110526ed2b8b36c769387

The updated Migration700005 will only error on this line so it should not happen for you again on this update
![image](https://github.com/user-attachments/assets/2488d9f9-955d-4a1e-91dd-e42512e4c8dc)

What we've done here is simplify the update with just plain sql instead of loading objects and causing possible errors

eee3952a5da7c9c07ba13e3a659487cd2e195dee has the latest version

**Reply by gpe5280** [reply]
The good news is the DB updates appear to have completed successfully.

![Selection_20250115_092149](https://github.com/user-attachments/assets/003f6148-4f91-4569-8cd2-9e6c1238e61e)

After clicking on "Return to main page" I'm presented with the login screen. It appears I log in successfully, but the screen is blank after logging in. DevTools reveals the following:

![Selection_20250115_093158](https://github.com/user-attachments/assets/0c8622aa-a05b-4f0a-a9d8-d0ab5bf2d982)

The Ampache log showed:

`2025-01-15T16:36:06+00:00 [gpe@javazen.com] (/home/websites/domains/audio.javazen.com/ampache-7.1.1_all_php8.3/src/Module/Util/Ui.php:367) -> Call to undefined function Ampache\Module\Util\simplexml_load_file()`

This issues was resolved by running...

`apt install php-xml`

...which installed...

`php-xml php8.3-xml`

No more blank screen and everything seems to be running fine. Looks like this particular issues has been resolved. Thank you very much for your help!

**Reply by lachlan-00** [reply]
awesome, i'm glad it was easy to solve!

Sorry for the xmas break but Amapche 7 is worth the effort!
